docs/components/byok/aws-kms-setup.md

```diff
@@ -0,0 +1,137 @@
+- BYOK keys cannot be rotated automatically by Camunda 8.8; any rotation must be done manually in AWS KMS.
+- Make sure the key policy grants your Camunda cluster the required access.
+  :::
+
+## Step 1: Create a Camunda 8 SaaS cluster
+
+1. Sign in to the [Camunda Console](https://console.camunda.io/).
+2. Navigate to the **Cluster** section and click **Create New Cluster**.
+3. For **Region**, select an AWS region where you want your cluster deployed.
+4. Under **Encryption at rest**, choose **External**.
+5. Click **Create Cluster**.
+
+:::note Screenshot placeholder
+[Insert screenshot of cluster creation page with "External" encryption selected]
+:::
+
+After creation, the Console will display a **unique Camunda Cluster IAM role ARN**. You will use this in the next step when configuring your KMS key.
+
+## Step 2: Create and configure a KMS key in AWS
+
+Follow these steps in your AWS account:
+
+1. Open the **AWS KMS console**.
+2. Click **Create key**.
+3. Choose **Symmetric** key type.
+4. Select **Region** matching your Camunda cluster.
+5. Click **Next: Key administrators** and assign at least one administrator.
+6. Click **Next: Key usage permissions**. Here, you need to grant your Camunda cluster role access.
+
+### Example KMS key policy
+
+Replace `<CAMUNDA_CLUSTER_ROLE_ARN>` with the IAM role ARN from Step 1:
+
+```json
+{
+  "Version": "2012-10-17",
+  "Id": "key-policy-camunda-byok",
+  "Statement": [
+    {
+      "Sid": "Enable IAM User Permissions",
+      "Effect": "Allow",
+      "Principal": {
+        "AWS": "<YOUR_AWS_ACCOUNT_ROOT_ARN>"
+      },
+      "Action": "kms:*",
+      "Resource": "*"
+    },
+    {
+      "Sid": "Allow Camunda Cluster Access",
+      "Effect": "Allow",
+      "Principal": {
+        "AWS": "<CAMUNDA_CLUSTER_ROLE_ARN>"
+      },
+      "Action": [
+        "kms:Encrypt",
+        "kms:Decrypt",
+        "kms:ReEncrypt*",
+        "kms:GenerateDataKey*",
+        "kms:DescribeKey"
+      ],
+      "Resource": "*"
+    }
+  ]
+}
+```
+
+:::warning Key policy warnings
+
+- Ensure you do not restrict the Camunda cluster IAM role from any required KMS actions; otherwise encryption will fail.
+- Keys must remain in the same region as your cluster.
+- Any revocation of access will immediately break cluster encryption.
+
+:::
+
+7. Click **Finish** to create the key.
+8. Copy the **Key ARN**; you will need it in the Camunda Console.
+
+:::note Screenshot placeholder
+[Insert screenshot of AWS KMS key details showing Key ARN]
+:::
+
+## Step 3: Associate KMS key with Camunda cluster
+
+1. Return to the Camunda Console.
+2. In the cluster details page, locate the KMS Key ARN input field.
+3. Paste your AWS KMS Key ARN from Step 2.
+4. Confirm and apply the key. The Camunda Operator will provision storage using your BYOK key for:
+   1. Document handling storage
+   2. Backup storage
+   3. Zeebe persistent disks
+   4. Elasticsearch persistent disks
```

Comment:
Not sure on the naming how we expose it. 

Maybe it makes sense to rename in some parts

Zeebe -> Camunda / OrchestrationCluster (?) 
Elasticsearch -> Secondary Storage (just maybe - not sure how we usually refer to it in SaaS docs).